Fixed inconsistency in Notification Number Box

miPlodder opened the pull request:
Fixes #537 

![_1](https://user-images.githubusercontent.com/22195621/35582775-c96a6cd2-0615-11e8-9127-a45f7b790f43.png)

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

miPlodder committed:
Fixed inconsistency in Notification Number Box

miPlodder commented:
@therajanmaurya @tarun0 Please review this PR.

dilpreet96 commented:
@miPlodder Good Catch!
But also add `getMvpView().showNotificationCount(0);`  to this line: https://github.com/openMF/mifos-mobile/blob/development/app/src/main/java/org/mifos/mobilebanking/presenters/HomeOldPresenter.java#L195

therajanmaurya commented:
@miPlodder Please open the new activity on click notification button instead of fragment transaction in home activity.

miPlodder commented:
@dilpreet96 @therajanmaurya Sorry, for delayed reply (had some issues with my system). I will make the required changes and send a PR.

miPlodder commented:
@therajanmaurya @dilpreet96 I have made the request changes and updated the PR. Please review the PR.

**Screenshot of Notification Activity** 

( I have added back button for improving User Experience )

![_1](https://user-images.githubusercontent.com/22195621/35821901-895ebb74-0ad0-11e8-95e6-72a868df43bc.png)

miPlodder commented:
@therajanmaurya @dilpreet96 I would like to add _**Toast message**_ in the Notification Activity, when there is no Notification for the User. (It will improve the **UX** and will be better than showing an empty recycler view).

If you think that it's a valuable addition, I would like to work on it.

dilpreet96 commented:
@miPlodder , Instead of adding a toast you can show a text in the middle saying `No Notifications`

dilpreet2028 commented on app/build.gradle:
```diff
@@ -66,6 +67,7 @@ dependencies {
 
     implementation fileTree(dir: 'libs', include: ['*.jar'])
 
+    implementation 'com.android.support.constraint:constraint-layout:1.0.2'
```
Are you using constraint layout anywhere? if no then remove it

miPlodder replied:
Ok (Added by Studio when I created new activity for Notification)

dilpreet2028 commented on app/build.gradle:
```diff
@@ -81,7 +83,7 @@ dependencies {
 
     //Square dependencies
-    implementation ('com.squareup.retrofit2:retrofit:$rootProject.retrofitVersion') {
+    implementation('com.squareup.retrofit2:retrofit:$rootProject.retrofitVersion') {
```
Add the space which you have removed

miPlodder replied:
Ok

dilpreet2028 commented on app/src/main/java/org/mifos/mobilebanking/ui/fragments/HomeOldFragment.java:
```diff
@@ -132,11 +133,16 @@ public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
-                ((BaseActivity) getActivity()).replaceFragment(NotificationFragment.newInstance(),
-                        true, R.id.container);
+
+                startActivity(
```
Please change this into 1 line or 2, no need of intending it in this way

miPlodder replied:
Ok

dilpreet2028 commented on app/src/main/AndroidManifest.xml:
```diff
@@ -1,10 +1,10 @@
 <?xml version="1.0" encoding="utf-8"?>
```
@miPlodder , why you have made these changes in manifest, please revert them and just add a new <activity> tag for the activity that you have created.

miPlodder replied:
Ok

dilpreet2028 commented on app/src/main/res/layout/fragment_notification.xml:
```diff
@@ -8,4 +8,13 @@
+        android:layout_height="match_parent"
+        android:gravity="center"
+        android:visibility="gone"
+        android:text="No Notifications"
```
@miPlodder , Don't use raw strings instead create a resource in strings.xml for this.

miPlodder replied:
I will make the changes and update the PR.

dilpreet2028 reviewed (changes requested):
@miPlodder also post a Screenshot of notifications screen.

miPlodder commented:
When activity has no notification,

![_old 1](https://user-images.githubusercontent.com/22195621/36161373-7d1c353c-1109-11e8-8235-5c49ebc2defe.png)

When activity has notification,

![_old 2](https://user-images.githubusercontent.com/22195621/36161376-7d6f90ce-1109-11e8-9cb0-3c7a3b9a469a.png)

miPlodder commented:
@dilpreet96 
I would like to add SwipeRefreshLayout to this activity to make activity consistent with other activities and it will surely enhance the UI/UX for the end user.

Should I add it in this PR ?

miPlodder commented on app/src/main/res/layout/action_bar_notification_layout.xml:
```diff
@@ -25,6 +25,7 @@
         android:textColor="@color/white"
         android:layout_height="wrap_content"
         android:gravity="top|end"
+        android:visibility="gone"
```
I have made a change here, this **TextView** was being show for a fraction of second, until we make it's visibility to **GONE** using Java 
(https://github.com/miPlodder/mifos-mobile/blob/7c93d583dbc8482a111943bdf64b32284a5194d6/app/src/main/java/org/mifos/mobilebanking/ui/fragments/HomeOldFragment.java#L289)

miPlodder commented:
@dilpreet96 I have updated the PR. SwipeRefreshLayout is added and made some changes in TextView as mentioned in the comment above.

Please review this PR.

dilpreet96 commented:
Good work @miPlodder
